challenge.rb

```diff
@@ -1,26 +1,39 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  # Use map to apply the String.capitalize
+  # method to each string in the array
+  input.map { |item| item.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  # Use select to filter for only items
+  # that have the string value "dog"
+  input.select { |item| item == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  # Use select to filter for only items
+  # that do not have the string value "dog"
+  input.select { |item| item != "dog" }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  # Use count to count the elements
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  # Use first to get the first two
+  # elements of the Enumerable
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  # Use select to filter for strings
+  # that have either "c" or "d" as
+  # their first character
+  input.select { |item|
+    item[0,1] == "c" or item[0,1] == "d" }
```

Comment:
Check out [`String#start_with?`](https://ruby-doc.org/core-2.6.1/String.html#method-i-start_with-3F) for a way to simplify this a bit.

Reply:
Will do, thanks!